src/protocol/abi/argument_encoding.md

```diff
@@ -0,0 +1,245 @@
+# Argument Encoding
+
+When crafting transaction script data, you must encode the arguments you wish to pass to the script.
+
+## Types
+
+These are the available types that can be encoded in the ABI:
+
+- Unsigned integers:
+  - `u8`, 8 bits.
+  - `u16`, 16 bits.
+  - `u32`, 32 bits.
+  - `u64`, 64 bits.
+- Boolean: `bool`, either `0` or `1` encoded identically to `u8`.
+- B256: `b256`, arbitrary 256-bits value.
+- Address : `address`, a 256-bit (32-byte) address.
+- Fixed size string
+- Array
+- Sum types
+- Structs
+
+These types are encoded in-place. Here's how to encode them. We define `enc(X)` the encoding of the type `X`.
+
+## Unsigned Integers
+
+`u<M>` where `M` is either 8, 16, 32, or 64 bits.
+
+`enc(X)` is the big-endian representation of `X` left-padded with zero-bytes. Total length must be 8 bytes.
+
+> **Note:** since all integer values are unsigned, there is no need to preserve the sign when extending/padding; padding with only zeroes is sufficient._
+
+**Example:**
+
+Encoding `42` yields: `0x000000000000002a`, which is the binary representation of the decimal number `42`, right-aligned to 8 bytes.
```

Comment:
OTOH, is 'big-endian' too nerdy and 'right aligned' better?  Right aligned sounds weird to me, but I'm a language nerd.

Reply:
I would prefer big-endian myself.